### PR TITLE
replace OpenAI with hugging face version #491

### PR DESCRIPTION
**Pull request by phu-pmh**
Replaced old Tensorflow version with hugging face's version. Tested using `config/spring19_seminar/cola_gpt.conf`, and it ran fine.

- Note: Need to download pre-trained model folder [here](https://github.com/openai/finetune-transformer-lm/tree/master/model) and place it under `src/openai_transformer_lm/pytorch_huggingface/` folder. (Should I add this to README?)

### CODE REVIEW

**Comment by phu-pmh** [comment]
Added info on open-ai setup in README.

**Comment by sleepinyourhat** [comment]
Taking a look...

File: README.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ To use the PRPN sentence encoder from [***Neural language modeling by jointly le
 
 We also include an experimental option to use a shared [Transformer](https://arxiv.org/abs/1706.03762) in place of the shared BiLSTM by setting ``sent_enc = transformer``. When using a Transformer, we use the [Noam learning rate scheduler](https://github.com/allenai/allennlp/blob/master/allennlp/training/learning_rate_schedulers.py#L84), as that seems important to training the Transformer thoroughly. 
 
-We also support using pretrained Transformer language models. To use the OpenAI transformer model, set `openai_transformer = 1`.
+We also support using pretrained Transformer language models. To use the OpenAI transformer model, set `openai_transformer = 1`, download the [model](https://github.com/openai/finetune-transformer-lm) folder that contains pre-trained models, and place it under `src/openai_transformer_lm/pytorch_huggingface/`.
```
Comment (sleepinyourhat) [inline README.md]: Why do we need people to manually download these? The huggingface code seems like it should be able to automatically download any needed weights (which is how we use BERT).

**Comment by phu-pmh** [comment]
> I'm surprised you needed to change so little. Is everything in defaults.conf correct?

Yes, I went through the codes yesterday. Huggingface uses the same module names and function names as OpenAI so I only had to change the import statements. `defaults.conf` is correct.

File: README.md (hunk 1 above)
Comment (phu-pmh) [reply README.md]: Huggingface does not automatically download the OpenAI models. They ask to manually download the OpenAI models in their README.

Comment (sleepinyourhat) [reply README.md]: Got it, thanks! The readme makes it sound possible, but looking at the code, I don't see any URL.

**Comment by sleepinyourhat** [comment]
Run a short test, maybe with demo.conf (with GPT turned on) and make sure you get the same better-than-chance results on some task with old GPT and new GPT. Then we can merge.

**Comment by phu-pmh** [comment]
> Run a short test, maybe with demo.conf (with GPT turned on) and make sure you get the same better-than-chance results on some task with old GPT and new GPT. Then we can merge.

I'll do that!

**Pull request merged**